src/envoy/alts/alts_socket_factory.cc

```diff
@@ -97,9 +103,8 @@ UpstreamAltsTransportSocketConfigFactory::createTransportSocketFactory(
         return std::make_unique<Security::TsiHandshaker>(handshaker,
                                                          dispatcher);
       },
-      [peers](const tsi_peer &peer, std::string &err) {
-        return doValidate(peer, peers, err);
-      });
+      // Skip validation if peers is empty.
+      peers.empty() ? empty_validator : actual_validator);
```

Comment:
you should be able to use `nullptr` here instead of empty_validator.

Reply:
It seems I have to use static_case to cast the nullptr to have the same type as actual_validator. I prefer not to use static_cast so refactored to not to use empty_validator.